filament/src/materials/ssao/bilateralBlurBentNormals.mat

```diff
@@ -0,0 +1,135 @@
+material {
+    name : bilateralBlurBentNormals,
+    parameters : [
+        {
+            type : sampler2dArray,
+            name : ssao,
+            precision: medium
+        },
+        {
+            type : float2,
+            name : axis,
+            precision: high
+        },
+        {
+            type : int,
+            name : sampleCount
+        },
+        {
+            type : float,
+            name : farPlaneOverEdgeDistance
+        },
+        {
+            type : float[16],
+            name : kernel
+        }
+    ],
+    outputs : [
+        {
+            name : aoData,
+            target : color,
+            type : float3
+        },
+        {
+            name : bnData,
+            target : color,
+            type : float3
+        }
+    ],
+    variables : [
+        vertex
+    ],
+    domain : postprocess,
+    depthWrite : false,
+    depthCulling : false
+}
+
+vertex {
+    void postProcessVertex(inout PostProcessVertexInputs postProcess) {
+        postProcess.vertex.xy = postProcess.normalizedUV;
+    }
+}
+
+fragment {
+    #include "ssaoUtils.fs"
+
+    void dummy(){}
+
+    float bilateralWeight(in highp float depth, in highp float sampleDepth) {
+        float diff = (sampleDepth - depth) * materialParams.farPlaneOverEdgeDistance;
+        return max(0.0, 1.0 - diff * diff);
+    }
+
+    void tapAO(const highp sampler2DArray saoTexture, highp vec2 uv,
+            out float ao, out highp float sampleDepth) {
+        vec3 data = textureLod(saoTexture, vec3(uv, 0.0), 0.0).rgb;
+        ao = data.r;
+        sampleDepth = unpack(data.gb);
+    }
+
+    void tapBN(const highp sampler2DArray saoTexture, highp vec2 uv,
+            out vec3 bentNormal) {
+        vec3 data = textureLod(saoTexture, vec3(uv, 1.0), 0.0).xyz;
+        bentNormal = unpackBentNormal(data);
+    }
+
+    void postProcess(inout PostProcessInputs postProcess) {
+        highp vec2 uv = variable_vertex.xy; // interpolated at pixel's center
+
+        float ao;
+        highp float depth;
+        highp float sampleDepth;
+        vec3 bn;
+
+        vec3 data = textureLod(materialParams_ssao, vec3(uv, 0.0), 0.0).rgb;
+        ao = data.r;
+        depth = unpack(data.gb);
+
+        if (data.g * data.b == 1.0) {
```

Comment:
Should it be >= 1.0 just in case?

Reply:
No it can't be > 1, since `.g` and `.b` are 0 to 1, also .`gb` encode a depth scaled between 0 and 1 and this encoding exists only because we're storing the depth in 8 bits channels.